amr/run.wdl

```diff
@@ -383,3 +389,55 @@ task ZipOutputs {
         docker: docker_image_id
     }
 }
+
+task MakeGeneCoverage {
+    input { 
+        File main_amr_results 
+        File main_output_json
+        String docker_image_id
+    }
+    command <<<
+    set -euxo pipefail
+    python3 <<CODE
+    import pandas as pd
+    import numpy as np
+    import json
+    df = pd.read_csv("~{main_amr_results}", delimiter="\t").loc[:, ["ORF_ID", "ID", "Model_ID", "Hit_Start", "Hit_End", "Percentage Length of Reference Sequence"]]
+
+    # create seq length reference map
+    with open("~{main_output_json}") as json_file:
+        rgi_main_json = json.load(json_file)
+    db_seq_length = {}
+    for ind, row in df.iterrows():
+        db_seq_length[row["Model_ID"]] = len(rgi_main_json[row["ORF_ID"]][row["ID"]]["dna_sequence_from_broadstreet"])
+
+    agg_res = df.groupby(["ID", "Model_ID"]).agg(lambda x: list(x))
+
+    gene_coverage = []
+    for ind, row, in agg_res.iterrows():
+      max_end = -1
+      gene_coverage_bps = 0
+      for start, end, in sorted(zip(row["Hit_Start"], row["Hit_End"]), key=lambda x: x[0]):
+        gene_coverage_bps += max(0, # if end-max(max_end, start) is negative
```

Comment:
Should calculate the total coverage breadth (sum of basepairs - overlapping regions) assuming an input sorted by HIt_Start. The cases I've considered are: 

1. intervals don't overlap
2. intervals partially overlap ( interval 2 ends after end of interval 1) 
3. intervals wholly overlap ( interval 2 is wholly contained in interval 1)